File: src/lib/commerce/sellerPlatform.test.ts
import { describe, it, expect } from 'vitest'
import { sellerPlatform } from './sellerPlatform'

describe('sellerPlatform — free-text seller → fixed enum', () => {
  it('buckets the known marketplaces case-insensitively', () => {
    expect(sellerPlatform('Shopee')).toBe('shopee')
    expect(sellerPlatform('shopee vietnam')).toBe('shopee')
    expect(sellerPlatform('Lazada')).toBe('lazada')
    expect(sellerPlatform('Tiki')).toBe('tiki')
    expect(sellerPlatform('Tiki Trading')).toBe('tiki')
    expect(sellerPlatform('TikTok Shop')).toBe('tiktok')
    expect(sellerPlatform('tiktok')).toBe('tiktok')
  })

  it('TikTok is not mis-bucketed as tiki', () => {
    expect(sellerPlatform('TikTok Shop')).not.toBe('tiki')
  })

  it('everything else — and empty/null — is "other" (never the raw string)', () => {
    expect(sellerPlatform('CellphoneS')).toBe('other')
    expect(sellerPlatform('Điện Máy Xanh')).toBe('other')
    expect(sellerPlatform('')).toBe('other')
    expect(sellerPlatform(null)).toBe('other')
    expect(sellerPlatform(undefined)).toBe('other')
  })
})

File: android/app/src/test/java/com/tappyai/app/chat/SellerPlatformTest.kt
package com.tappyai.app.chat

import org.junit.Assert.assertEquals
import org.junit.Test

/** The Kotlin twin of web's sellerPlatform — a free-text seller maps to a fixed enum, and the
 *  raw string is never surfaced. Mirrors src/lib/commerce/sellerPlatform.test.ts. */
class SellerPlatformTest {

    @Test
    fun `known marketplaces bucket case-insensitively`() {
        assertEquals("shopee", sellerPlatform("Shopee"))
        assertEquals("shopee", sellerPlatform("shopee vietnam"))
        assertEquals("lazada", sellerPlatform("Lazada"))
        assertEquals("tiki", sellerPlatform("Tiki"))
        assertEquals("tiki", sellerPlatform("Tiki Trading"))
        assertEquals("tiktok", sellerPlatform("TikTok Shop"))
    }

    @Test
    fun `TikTok is not mis-bucketed as tiki`() {
        assertEquals("tiktok", sellerPlatform("TikTok Shop"))
    }

    @Test
    fun `everything else and null is other`() {
        assertEquals("other", sellerPlatform("CellphoneS"))
        assertEquals("other", sellerPlatform("Điện Máy Xanh"))
        assertEquals("other", sellerPlatform(""))
        assertEquals("other", sellerPlatform(null))
    }
}
